### ISSUE TITLE
[Update agent key] Cells need to leave networks and join new networks

### ISSUE THREAD

**Issue by jost-s** [holochain/holochain#4040]
AC:
* During agent update process, while app is disabled, leave() all old agents and join() all new agents in parallel
* Don't re-enable app until joining is complete for all cells under the new agent
* Test that
  * ops by the old agent can be published by the new agent
  * ops by the new agent can be published by the new agent

**Comment by jost-s** [comment]
Duplicate of #3647
